subsys/net/ip/net_if.c

```diff
@@ -4196,6 +4205,65 @@ int net_if_up(struct net_if *iface)
 	return status;
 }
 
+int net_if_connect(struct net_if *iface)
+{
+	int status = 0;
+
+	NET_DBG("iface %p connect", iface);
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	if (!net_if_flag_is_set(iface, NET_IF_UP)) {
+		status = -ESHUTDOWN;
+		goto out;
+	}
+
+	/* Question to reviewers:
+	 * Would it be cleaner to attach the underlying API implementation to the
+	 * net_l2 struct instead...? This would require modifications be made to allow
+	 * offloaded sockets to have L2 implementations. But I think we will need this
+	 * change anyways, eventually, for other reasons, so maybe now is the time?
+	 */
+
+	const struct net_if_api *api = net_if_get_device(iface)->api;
+
+	if (api->connect) {
```

Comment:
If `api->connect` is null should the status be still 0?

Reply:
I go back and forth on whether these functions should yield ENOTSUP, or just no-op for ifaces that don't have a concept of connectivity.

Probably better to return ENOTSUP, and then have `conn_mgr` intelligently handle the error, or otherwise detect whether the iface supports connectivity